pages/[envId]/products/[slug].tsx

```diff
@@ -21,12 +22,23 @@ type Props = Readonly<{
 
 interface IParams extends ParsedUrlQuery {
   slug: string
+  envId: string
 }
 
 export const getStaticPaths: GetStaticPaths = () => {
-  return getProductItemsWithSlugs()
+  const envId = process.env.NEXT_PUBLIC_KONTENT_ENVIRONMENT_ID;
+  if (!envId) {
+    throw new Error("Missing 'NEXT_PUBLIC_KONTENT_ENVIRONMENT_ID' environment variable.");

```

Comment:
It is a bit unfortunate, that this code needs to be copy-pasted into every `getStaticPaths` and the code that gets the envId from the url needs to be copy-pasted into every `getStaticProps`.